packages/collector/src/types/shared.d.ts

```diff
@@ -2,7 +2,7 @@ import { CollectorConfig } from './collector';
 import { GenericLogger, InstanaBaseSpan } from '@instana/core/src/core';
 
 export interface Init {
-  currentSpan(): InstanaBaseSpan;
+  currentSpan(): { span: InstanaBaseSpan };
```

Comment:
I understand what you mean in general, but the differentiation if the span is available happens on runtime and should not affect the TS type checking. 

If you are running into a TS error, please share here with us 👍

Reply:
Thanks for acknowledging.
I'm in agreement, it does not affect TS type checking - TS would indicate the code should run without any type errors.
This is not necessarily the truth if there is misalignment between the type definition and the actual JS implementation.

Here's an example of a code snippet that would produce a runtime error, without producing a warning / error during `tsc build`.

The TypeScript compiler has no issue with this code as it does not know the value can actually be undefined at runtime and therefore does not warn the developer during `tsc build`.

```ts
import { type InstanaBaseSpan } from '@instana/core/src/core.js';
import instana from '@instana/collector';

// Re-usable function to get subset of fields from the span to use through-out the application.
function getLogDetailFromSpan(span: InstanaBaseSpan) {
  return span.t; // TypeError: Cannot read properties of undefined (reading 't')
}

// Elsewhere in the application as part of handling an HTTP request.
// Explicitly specifying the type here, but this is what TS would derive from the current version anyway.
const currentSpan: { span: InstanaBaseSpan } = instana.currentSpan();
const traceId = getLogDetailFromSpan(currentSpan);
```
The code produces an error at runtime when the collector is not initialized (which is the case for me during local development) or when the current call isn't traced by the collector.

Reply:
Your code snippet is not right.

> instana.currentSpan();

Either returns a SpanHandle or NoopSpanHandle.

> NoopSpanHandle {} or SpanHandle { span }

You could protect your code like this:

```ts
...

const spanHandle = instana.currentSpan();

// Option 1
if (spanHandle.constructor.name === 'NoopSpanHandle') {}

// Option 2
if (!spanHandle.span) {}
```

Keep in mind: **There should be always an active span in an async context.**

```
app.get('something', (req) => {
   const span = instana.currentSpan();
   // span should exist
})
```

> Either returns a SpanHandle or NoopSpanHandle.

We definitely can improve our SDK. It would be much better if customers could simply do `!instana.currentSpan()`. A PR is welcome to improve this without breaking the old notation.

Reply:
@legehwahn See also https://github.com/instana/nodejs/pull/1374. Another improvement, which makes it easier to access the information of a span.